src/main/java/org/embulk/output/mailchimp/MailChimpHttpClient.java

```diff
@@ -0,0 +1,160 @@
+    private final ObjectMapper jsonMapper = new ObjectMapper()
+            .configure(com.fasterxml.jackson.core.JsonParser.Feature.ALLOW_UNQUOTED_CONTROL_CHARS, false)
+            .configure(com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    private Jetty92RetryHelper retryHelper;
+
+    /**
+     * Instantiates a new Mailchimp http client.
+     *
+     * @param task the task
+     */
+    public MailChimpHttpClient(MailChimpOutputPluginDelegate.PluginTask task)
+    {
+        retryHelper = createRetryHelper(task);
+    }
+
+    /**
+     * Close @{@link Jetty92RetryHelper} connection
+     */
+    public void close()
+    {
+        if (retryHelper != null) {
+            retryHelper.close();
+        }
+    }
+
+    public JsonNode sendRequest(final String endpoint, final HttpMethod method,
+                                final MailChimpOutputPluginDelegate.PluginTask task)
+    {
+        return sendRequest(endpoint, method, "", task);
+    }
+
+    public JsonNode sendRequest(final String endpoint, final HttpMethod method, final String content,
+                                final MailChimpOutputPluginDelegate.PluginTask task)
+    {
+        final String authorizationHeader = getAuthorizationHeader(task);
+
+        try {
+            String responseBody = retryHelper.requestWithRetry(
+                    new StringJetty92ResponseEntityReader(task.getTimeoutMillis()),
+                    new Jetty92SingleRequester()
+                    {
+                        @Override
+                        public void requestOnce(HttpClient client, Response.Listener responseListener)
+                        {
+                            Request request = client
+                                    .newRequest(endpoint)
+                                    .accept("application/json")
+                                    .method(method);
+                            if (method == HttpMethod.POST || method == HttpMethod.PUT) {
+                                request.content(new StringContentProvider(content), "application/json");
+                            }
+
+                            if (!authorizationHeader.isEmpty()) {
+                                request.header("Authorization", authorizationHeader);
+                            }
+                            request.send(responseListener);
+                        }
+
+                        @Override
+                        public boolean isResponseStatusToRetry(Response response)
+                        {
+                            int status = response.getStatus();
+                            return status == 429 || status / 100 != 4;
+                        }
+                    });
+
+            return responseBody != null && !responseBody.isEmpty() ? parseJson(responseBody) : MissingNode.getInstance();
+        }
+        catch (HttpResponseException ex) {
```

Comment:
`HttpResponseException` should be not here. Its redundant. Then I will remove it.